shared/middleware/channels_api.rb

```diff
@@ -47,19 +47,34 @@ class ChannelsApi < Sinatra::Base
   #
   # Create a channel.
   #
+  # Optional query string param: ?src=<src-channel-id> creates the channel as
+  # a copy of the given src channel.
+  #
   post '/v3/channels' do
     unsupported_media_type unless request.content_type.to_s.split(';').first == 'application/json'
     unsupported_media_type unless request.content_charset.to_s.downcase == 'utf-8'
 
-    begin
-      data = JSON.parse(request.body.read)
-    rescue JSON::ParserError
-      bad_request
+    src_channel = request.GET['src']
+    storage_app = StorageApps.new(storage_id('user'))
```

Comment:
You can already spam `/v3/channels` with data and create an unlimited number of channels (there may be some limits per user, I'm not sure).  I guess this makes us a little more open to abuse because you can upload a single large channel once, and cause it to be cloned many times with a very small request.  Thoughts @davidsbailey?